### ISSUE TITLE
Interactive directory history

### ISSUE THREAD

**Issue by horeah** [horeah/PyCmd#7]
In order to allow for a longer directory history PyCmd should use an interactive, searchable Window (similar to the one used for zsh-style completions) when Alt-D is pressed.

**Comment by horeah** [comment]
A prototype implementation can be checked out on branch `new_dirhist`

**Comment by horeah** [comment]
@ufo You had some interesting ideas related to the directory history -- from what I can tell from your https://github.com/ufo/pycmd; how do you feel about this proposal?

In my view, the main advantages compared to the current directory history are:
 1. transient "window"-like behavior avoids polluting the screen content
 2. incremental search allows for a longer history (the current proof-of-concept uses 30)
 3. remove code and complexity
 
The biggest disadvantage is that the "jump-to-number" functionality is no longer supported -- we could maybe introduce support for Alt-N in the general Window class?

**Comment by ufo** [comment]
I just tried it out and I really think it is awesome! And I'm sure it would also make my own approach with a second fixed dir history (favorites) unnecessary, once finding an every day project folder is easy by starting to type its name.
Regarding the disadvantage. To be honest I tried the ALT-N Feature a few times in the past but could never really get used to it. I always felt comfortable enough navigating it with UP and DOWN keys. So personally I wouldn't miss anything. But of course I can only speak for myself.
